### PR TITLE
Add DeleteMoneyRequest

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
Introduces the ability to delete requests

cc @mountiny @Julesssss will you please review?

### Fixed Issues
$ https://github.com/Expensify/App/issues/19066

### Tests
Should be tested with [Auth](https://github.com/Expensify/Auth/pull/8108) and [Web-E](https://github.com/Expensify/Web-Expensify/pull/37764) PRs

**Scenario 1: Deleting from IOU report**
1. Request money from another user two times
2. Tap the report preview to navigate to the IOU report
3. Hover on the request preview for the 1st request and verify that you can see the `Trashcan` icon
4. Tap the icon and delete the 1st request
5. Verify that the 1st request is deleted
6. Do the same for the 2nd request
7. Verify that the iou report is deleted as well and you are navigated to the chat report
8. Verify that the report preview in the chat report was also deleted
9. Request money again
10. Navigate to the IOU report
11. Send a comment
12. Delete the only transaction on the report
13. Verify that the IOU report is not deleted and the total is $0

**Scenario 2: Deleting from transaction view**
1. Request money from another user two times
2. Tap the report preview to navigate to the IOU report
3. Tap on the 1st request preview to navigate to the 1st request thread
4. From the three dot menu, delete the transaction
5. Verify that the transaction and thread are deleted and you are navigated back to the IOU report
6. Tap on the remaining request preview to navigate to the request thread
7. Send a comment in the thread
8. From the three dot menu, delete the transaction
14. Verify that the header is updated to show only `[Deleted request]`
15. Navigate to the IOU report
16. Verify that the request preview now shows `[Deleted request]` and `1 reply`
17. Verify that the IOU total is $0
18. Navigate to the chat report
19. Verify that the report preview is still there and total is $0

- [ ] Verify that no errors appear in the JS console

### Offline tests
Same as test steps, but performed offline

### QA Steps
Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/22219519/c11e1fcd-4384-46c6-a67a-4363a56c4679

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/22219519/8167b0e7-8ebb-43d4-adf7-4b297d34bc88

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/22219519/a45dbfdb-b2fd-4f6d-a33c-557f76d246a7

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/22219519/4485d065-f361-47f7-a470-06b9fbbc739a

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/22219519/ee980165-a981-4c1a-a055-06392c71f78e

</details>

### CODE REVIEW

**Review by Julesssss — COMMENTED** [review]
1) Oh, the user owes $0.00 is a bit odd, are we improving that further? 

2) After the reportPreview disappears and then I make a new request, the Preview shows no text (though possibly unrelated)
Screenshot 2023-06-08 at 10 38 35

File: src/libs/IOUUtils.js [hunk 1]
```diff
@@ -32,10 +32,10 @@ function calculateAmount(numberOfParticipants, total, isDefaultUser = false) {
  * @param {String} actorEmail
  * @param {Number} amount
  * @param {String} currency
- * @param {String} type
+ * @param {String} isDeleting - whether the user is deleting the request
  * @returns {Object}
  */
-function updateIOUOwnerAndTotal(iouReport, actorEmail, amount, currency, type = CONST.IOU.REPORT_ACTION_TYPE.CREATE) {
+function updateIOUOwnerAndTotal(iouReport, actorEmail, amount, currency, isDeleting = false) {
```
Comment (Julesssss) [inline src/libs/IOUUtils.js:38]: Curious why isDeleting is preferred to just passing the type?

File: src/components/MoneyRequestHeader.js [hunk 2]
```diff
@@ -90,6 +94,22 @@ const MoneyRequestHeader = (props) => {
...
+        IOU.deleteMoneyRequest(parentReportAction.originalMessage.IOUTransactionID, parentReportAction, true);
+    }, [parentReportAction]);
+
+    if (props.isSingleTransactionView && isDeletedParentAction) {
+        return (
+            <HeaderView
+                reportID={props.report.reportID}
+                onNavigationMenuButtonClicked={() => Navigation.goBack(ROUTES.HOME)}
+                personalDetails={props.personalDetails}
+                report={props.report}
```
Comment (mountiny) [inline src/components/MoneyRequestHeader.js:108]: Why do we need to pass both, report and reportID?

**Review by mountiny — COMMENTED** [review]
In general code looks good, but it was quite hard to test withouth the backend and I saw some weird things also given the report preview is not optimistic, I would wait for the web abd wuth to be ready to test the full flow.

File: src/libs/IOUUtils.js (hunk 1 above)
Comment (luacmartins) [reply src/libs/IOUUtils.js:38]: no reason, i just thought it was clearer to pass a bool and avoid the comparison.

File: src/components/MoneyRequestHeader.js (hunk 2 above)
Comment (luacmartins) [reply src/components/MoneyRequestHeader.js:108]: Good point. This was copy/pasta from ReportScreen. I removed it in both. Thanks!

File: src/libs/actions/IOU.js [hunk 3]
```diff
@@ -674,110 +669,198 @@ function splitBillAndOpenReport(participants, currentUserLogin, amount, comment,
...
-            reportActionID: optimisticIOUAction.reportActionID,
-            iouReportID: updatedIOUReport.reportID,
+            reportActionID: reportAction.reportActionID,
         },
         {optimisticData, successData, failureData},
     );
 
-    if (shouldCloseOnDelete) {
-        Navigation.dismissModal(iouReportID);
+    // STEP 7: Navigate the user depending on which page they are on and which resources were deleted
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/libs/actions/IOU.js:856]: If you delete the request from the IOU page and the IOU report is being deleted, we still need to navigate you to the chat report.

**Comment by luacmartins** [comment]
>Oh, the user owes $0.00 is a bit odd, are we improving that further?

Asked in thread with Jason and Tom

>After the reportPreview disappears and then I make a new request, the Preview shows no text (though possibly unrelated)

I think this might be a weird behavior because the API is not working yet. I think a code review for the general approach was enough for now and it seems like we agree on the approach. I'll work on the API now and we should test this once that's ready.

File: src/libs/actions/IOU.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mountiny) [reply src/libs/actions/IOU.js:859]: That is only in the case the request was a last request and there was also no comment on the report or on the request, right?

[the remark this answers is not in the document]
Comment (luacmartins) [reply src/libs/actions/IOU.js:859]: Correct

**Comment by luacmartins** [comment]
This is ready for review!

**Comment by luacmartins** [comment]
@0xmiroslav no, there are both still in review! You can review the code, but I'd hold testing until those PRs are deployed.

**Comment by mountiny** [comment]
Got this crash when deleting from the money 
```
Uncaught TypeError: Cannot read properties of undefined (reading '0')
    at Module.deleteMoneyRequest (IOU.js:669:1)
    at Object.eval [as onSelected] (MoneyRequestHeader.js:67:5)
    at Object.onModalHide (index.js:74:1)
    at Object.hideModal [as onModalHide] (index.web.js:25:1)
    at BaseModal.hideModal (BaseModal.js:87:1)
    at ReactNativeModal.eval (modal.js:427:1)
    at callCallback (react-dom.development.js:13122:1)
    at commitUpdateQueue (react-dom.development.js:13143:1)
    at commitLayoutEffectOnFiber (react-dom.development.js:23263:1)
    at commitLayoutMountEffects_complete (react-dom.development.js:24578:1)
```

Got this on some older request from the account, this is odd
```
UUID
: 
"dd799068-2815-4d79-b81a-aa7e91545f78"
code
: 
666
data
: 
[]
htmlMessage
: 
""
jsonCode
: 
404
message
: 
"You do not have the permissions required to delete expenses from this report."
onyxData
: 
[]
title
: 
"Cannot Remove Expenses"
type
: 
"Expensify\\Error\\Transaction\\CantReportTransactions"
```

**Comment by mountiny** [comment]
When you delete a request which has comment on it so the thread stays around, then the header of the transaction has darker green colour
image

**Comment by mountiny** [comment]
If I got a comment on the iou report where I deleted all the requests it stays in the history looking like this 
image, should we be hiding it somehow in this scenario?

**Comment by luacmartins** [comment]
lolwut? the header should be gone then 🤔 Anyways, keep them coming and I'll look into fixing these asap

**Comment by mountiny** [comment]
https://github.com/Expensify/Auth/pull/8108#pullrequestreview-1523132866

**Comment by luacmartins** [comment]
Gonna start looking at these issues tomorrow:

1. I couldn't reproduce the crash. Do you have any logs that can help?
2. For the header, I see that we made it scrollable [here](https://github.com/Expensify/App/pull/21918). So I'll have to address the new changes in my PR.
3. Not sure what the expected behavior for the $0 reportPreview is. @JmillsExpensify @trjExpensify should we hide the report preview if the total is $0 but we have comments in the IOU/Expense report?

**Comment by mountiny** [comment]
> I couldn't reproduce the crash. Do you have any logs that can help?

Unfortunately no, I dont have anything saved

**Comment by luacmartins** [comment]
There are two known bugs to me:

1. We can add duplicated report preview actions if we delete the last request on a report. This can also happens on main if the report total is $0 though
2. If the transaction thread is not being deleted, the header is still visible - this is a side-effect of what seems to be a navigation issue on main too. We're investigating in https://github.com/Expensify/App/issues/22909

**Comment by trjExpensify** [comment]
>Not sure what the expected behavior for the $0 reportPreview is. @JmillsExpensify @trjExpensify should we hide the report preview if the total is $0 but we have comments in the IOU/Expense report?

The report doesn't get deleted and new requests get added to that same report, right? If so, I would think it just remains in the chat history like that as a $0.00 report preview with the comment count. It's the same premise as how the actual iou/expenseReport would show in the LHN as "Tom owes $0.00" if you were to navigate to it for the comments we preserved. @JmillsExpensify, do you agree?

**Comment by mountiny** [comment]
when deleting the older request getting this error:

image

Getting these errors from that account:
```
?api? [hmmm] Throw ExpException - 8cfb80437a4dad76e6da8c403e8eb138 ~~ message: '404 payerAccountID and policyID not found' exceptionMessage: 'Auth CreateIOUTransaction returned an error' exceptionFile: '/vagrant/Web-Expensify/lib/Auth.php' exceptionLine: '124' exceptionCode: '404'
```

I think this will be the same issue with how the iouReportID is updated or the email is not share for users that are not known is added for as is seen on main

**Comment by mountiny** [comment]
The transaction thread is showing incorrect order manager/ submitter 
image

As you can see the account `test+navigation@test.com` has requested the money 

image

but when I go to the transaction detail its reversed and I cannot delete it even if I am the one who requested the money 

image

**Comment by mountiny** [comment]
I have deleted the request from transactions thread and was navigated to the iou report, but the request preview was still there until I have refreshed

**Comment by mountiny** [comment]
BTW figure the negative value of the iou report action (which is wrong) comes from here https://github.com/Expensify/Auth/blob/3332e12ceb7f9f723dcba51460c406e1bdbc90d4/auth/lib/Report.cpp#L3807-L3812 but I dont undesrstand fully why its like that

**Comment by luacmartins** [comment]
Tests are passing now 🎉 Gonna look into the reported bugs in the App PR to see if any of them come from Web.

**Comment by luacmartins** [comment]
>1. Error when deleting old request

I was not able to reproduce this error. Do you have detailed steps?

>2. The transaction thread is showing incorrect order manager/ submitter

I'm also not able to reproduce this issue. The header is correct to me. Do you have any more detailed steps? Are these old accounts?

>3. IOU preview is still visible when deleting from transaction thread?

I couldn't reproduce this because I found 2 other bugs.

1. App crashed when deleting from transaction thread. I pushed a fix for this already.
2. Despite [this line](https://github.com/Expensify/App/blob/64e048644a1077e1c82bbbf8565eb11452e83829/src/libs/actions/IOU.js#L974) returning the correct route, I keep being redirected to my chat with Concierge. Gonna look into this next.

**Comment by luacmartins** [comment]
So for 3.2, the redirect to the Concierge chat is coming from [here](https://github.com/Expensify/App/blob/64e048644a1077e1c82bbbf8565eb11452e83829/src/pages/home/ReportScreen.js#L183). It seems like we have logic in place to redirect to Concierge if a report is removed. Gonna look at a way to circumvent that.

**Review by Julesssss — COMMENTED** [review]
Tests are failing for me [here](https://github.com/Expensify/Web-Expensify/pull/37764#pullrequestreview-1536806863)

**Comment by mountiny** [comment]
Tried to delete old request and got this error:

```
2023-07-19T16:40:10.438801+00:00 expensidev2004 php-fpm: E0Iqmx /api.php test+20230614e@test.com !ecash1.3.42-20! ?api? [hmmm] Throw ExpException - ea7eac0b91b1ca51b6e612ecce75933c ~~ message: '404 Comment Not Found' exceptionMessage: 'Auth DeleteMoneyRequest returned an error' exceptionFile: '/vagrant/Web-Expensify/lib/Auth.php' exceptionLine: '124' exceptionCode: '404'
2023-07-19T16:40:10.438821+00:00 expensidev2004 php-fpm: E0Iqmx /api.php test+20230614e@test.com !ecash1.3.42-20! ?api? [info] [trace] StackTrace start
2023-07-19T16:40:10.438828+00:00 expensidev2004 php-fpm: E0Iqmx /api.php test+20230614e@test.com !ecash1.3.42-20! ?api? [info] [trace] 0/2. file '/vagrant/Web-Expensify/lib/Auth.php' function '_standardResponseHandler' line '2672'
2023-07-19T16:40:10.438833+00:00 expensidev2004 php-fpm: E0Iqmx /api.php test+20230614e@test.com !ecash1.3.42-20! ?api? [info] [trace] 1/2. file '/vagrant/Web-Expensify/lib/ReportAPI.php' function 'deleteMoneyRequest' line '2186'
2023-07-19T16:40:10.438838+00:00 expensidev2004 php-fpm: E0Iqmx /api.php test+20230614e@test.com !ecash1.3.42-20! ?api? [info] [trace] 2/2. file '/vagrant/Web-Expensify/api.php' function 'deleteMoneyRequest' line '914'
2023-07-19T16:40:10.438842+00:00 expensidev2004 php-fpm: E0Iqmx /api.php test+20230614e@test.com !ecash1.3.42-20! ?api? [info] [trace] StackTrace end
2023-07-19T16:40:10.438926+00:00 expensidev2004 php-fpm: E0Iqmx /api.php test+20230614e@test.com !ecash1.3.42-20! ?api? [warn] [OOPS!] API Threw: Auth DeleteMoneyRequest returned an error ExpException - ea7eac0b91b1ca51b6e612ecce75933c ~~ message: '404 Comment Not Found' exceptionMessage: 'Auth DeleteMoneyRequest returned an error' exceptionFile: '/vagrant/Web-Expensify/lib/Auth.php' exceptionLine: '124' exceptionCode: '404'
```

Also attaching a video where it was happening.

There was couple of issues but I dont think they are necessarily related to the Web changes. 

1. notice as the transaction thread report has automatically redirecte to Concierge? It happened three times I think, I did not click anything, could not see anything in console either
2. When clicking parent of the transaction thread it lead to the DM and not to the IOU report as it should - for the first time. Maybe related to the issues we have seen before.
3. Deleting failed with the error above
4. It seems to work fine for new requests so maybe we just need to test on some older requests if there is anything weird with them making this break
5. When user is in the iou report and the last transaction is removed, they are navigated to the concierge report as default, I think we should always navigate to the parent report in these case.

**Comment by mountiny** [comment]
@0xmiroslav This can be tested by c+ 🎉 against staging

**Comment by luacmartins** [comment]
OK, I went through the comments and here are the results:

@Julesssss 

>Verify that the iou report is deleted as well and you are navigated to the chat report

I didn't see any issues here on my end. Would you mind trying again and maybe record a video/getting logs if it still happens for you?

> The IOUPreview didn’t disappear from either the LHN or report

Same here.

@mountiny 

>notice as the transaction thread report has automatically redirecte to Concierge? It happened three times I think, I did not click anything, could not see anything in console either
>When user is in the iou report and the last transaction is removed, they are navigated to the concierge report as default, I think we should always navigate to the parent report in these case.

This is an issue and out of scope for this PR. We're working on it in [this thread](https://expensify.slack.com/archives/C04878MDF34/p1689871481652229)

>When clicking parent of the transaction thread it lead to the DM and not to the IOU report as it should - for the first time. Maybe related to the issues we have seen before

It seems like the link from the transaction thread to the parent IOU report is completely gone on main. So I'll say this is NAB for this PR and we can tackle that in a separate PR.

> Deleting failed with the error above. 
>It seems to work fine for new requests so maybe we just need to test on some older requests if there is anything weird with them making this break

I'm not able to reproduce this error even on older transactions. See video below for transactions from May 19-24. Maybe these transactions have some weird data saved in your local env?

https://github.com/Expensify/App/assets/22219519/9819c6ec-d1f2-4608-ae43-cdf075a632f8

**Comment by mountiny** [comment]
@0xmiroslav @aimane-chnaif this is ready to be tested.

**Comment by Julesssss** [comment]
@luacmartins unfortunately today I'm having to deal with recreating my vm. If the PR is still open next week I can hopefully retest. Lets assume for now it was an issue on my end though.

**Comment by luacmartins** [comment]
Thanks for the tests. Gonna try to reproduce and address those now. Although I'm inclined to address them in a follow up PR since this one is getting pretty big already. I'm gonna report back soon.

**Comment by luacmartins** [comment]
Bugs 1-5 and the offline issues come from the backend. I created [this draft PR](https://github.com/Expensify/Web-Expensify/pull/38305) to address them.

I don't think we should hold this PR on those fixes since they are small polish items.

**Comment by luacmartins** [comment]
@0xmiroslav can you give this one another review? I'll address the issues you [pointed out](https://github.com/Expensify/App/pull/19095#issuecomment-1645807063) in another PR, but I don't think we should block this one.

**Comment by JmillsExpensify** [comment]
@0xmiroslav Heads up, this is one of our most high-value PR reviews at the moment. Thank you!

**Comment by luacmartins** [comment]
@aimane-chnaif yes, it supports expense requests too. I suspect what you're saying is the same as what @0xmiroslav mentioned [here](https://github.com/Expensify/App/pull/19095#issuecomment-1645807063) and I'm addressing that in a separate PR.

**Comment by luacmartins** [comment]
Thanks! I'll address those as follow ups.

**Comment by mountiny** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by mountiny** [comment]
Ok I think we can go ahead and merge this, it was tested many time across platforms and we are aware of shortcomings we will nail in follow up PRs.

@luacmartins Can you make sure you keep an eye on `#expensify-bugs` to make sure we are not making duplicates for the issues we already know?

**Comment by luacmartins** [comment]
Will do!

**Comment by mountiny** [comment]
@luacmartins can you add this one to the follow ups? https://github.com/Expensify/App/pull/19095#issuecomment-1652212296

**Comment by luacmartins** [comment]
Added that issue to the follow ups and created a bunch of notification alerts for `#expensify-bugs`

**Review by mountiny — APPROVED** [review]
Great job on this one

**Comment by mountiny** [comment]
Everything was passing

**Comment by roryabraham** [comment]
Friendly piece of feedback that I think it would've been good to add some automated tests for this PR to the existing `IOUTest` test suite

File: src/libs/actions/IOU.js [hunk 4]
```diff
@@ -779,110 +772,213 @@ function splitBillAndOpenReport(participants, currentUserLogin, currentUserAccou
...
-        moneyRequestAction.originalMessage.comment,
-        [],
-        transactionID,
-        '',
-        iouReportID,
-    );
+    // STEP 2: Decide if we need to:
+    // 1. Delete the transactionThread - delete if there are no visible comments in the thread
+    // 2. Update the iouPreview to show [Deleted request] - update if the transactionThread exists AND it isn't being deleted
+    const shouldDeleteTransactionThread = transactionThreadID ? ReportActionsUtils.getLastVisibleMessage(transactionThreadID).lastMessageText.length === 0 : false;
```
Comment (allroundexperts) [inline src/libs/actions/IOU.js:794]: We missed to take in to account that there could be `MODIFIEDEXPENSE` actions which would make the last visible message as non-empty, and thus, prevent the thread from getting deleted.

File: src/libs/ReportUtils.js [hunk 5]
```diff
@@ -259,6 +259,10 @@ function isCurrentUserSubmitter(reportID) {
  * @returns {Boolean}
  */
 function canDeleteReportAction(reportAction, reportID) {
+    const isActionOwner = reportAction.actorAccountID === currentUserAccountID;
+    if (isActionOwner && ReportActionsUtils.isMoneyRequestAction(reportAction) && !isSettled(reportAction.originalMessage.IOUReportID)) {
```
Comment (s77rt) [inline src/libs/ReportUtils.js:263]: In addition to checking if the report is not settled we should have also checked if the report is submitted. In paid policies you can't delete a money request in a submitted report (unless you are the admin).

**Pull request merged**